lib/pbench/test/unit/server/test_user_auth.py

```diff
@@ -290,13 +279,37 @@ def test_update_user(client, server_config):
             )
             data = response.json
             assert response.status_code == 200
-            assert data["data"]["first_name"] == "firstname"
-            assert data["data"]["email"] == "user@domain.com"
+            assert data["first_name"] == "firstname"
+            assert data["email"] == "user@domain.com"
             time.sleep(1)
             response = login_user(client, server_config, "username", "newpass")
-            data_login = response.json
             assert response.status_code == 200
-            assert data_login["message"] == "Successfully logged in."
+
+    @staticmethod
+    def test_external_token_update(client, server_config):
+        """ Test for external attempt at updating auth token"""
+        with client:
+            resp_register = register_user(
+                client,
+                server_config,
+                username="username",
+                firstname="firstname",
+                lastname="lastName",
+                email="user@domain.com",
+                password="12345",
+            )
+            assert resp_register.status_code == 201
+
+            response = login_user(client, server_config, "username", "12345")
+            assert response.status_code == 200
+            data_login = response.json
+
+            response = client.put(
+                f"{server_config.rest_uri}/user/username",
+                json={"auth_tokens": "external_auth_token"},
+                headers=dict(Authorization="Bearer " + data_login["auth_token"]),
+            )
+            assert response.status_code == 400
```

Comment:
Would it be worthwhile to test the error message text?  (E.g., to confirm that it failed for the reason that it was supposed to fail for?)

Reply:
We could do that